File: src/main/java/com/eager/questioncloud/question/QuestionReviewEntity.java
package com.eager.questioncloud.question;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@Table(name = "question_review")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class QuestionReviewEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private Long questionId;

    @Column
    private Long reviewerId;

    @Column
    private String comment;

    @Column
    private int rate;

    @Column
    private LocalDateTime createdAt;

    @Column
    private Boolean isDeleted;

    @Builder
    public QuestionReviewEntity(Long id, Long questionId, Long reviewerId, String comment, int rate, LocalDateTime createdAt, Boolean isDeleted) {
        this.id = id;
        this.questionId = questionId;
        this.reviewerId = reviewerId;
        this.comment = comment;
        this.rate = rate;
        this.createdAt = createdAt;
        this.isDeleted = isDeleted;
    }
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/review/infrastructure/entity/QuestionReviewEntity.kt
package com.eager.questioncloud.core.domain.review.infrastructure.entity

import com.eager.questioncloud.core.domain.review.model.QuestionReview
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "question_review")
class QuestionReviewEntity private constructor(
    @GeneratedValue(strategy = GenerationType.IDENTITY) @Id private var id: Long?,
    @Column private var questionId: Long,
    @Column private var reviewerId: Long,
    @Column private var comment: String,
    @Column private var rate: Int,
    @Column private var createdAt: LocalDateTime,
    @Column private var isDeleted: Boolean
) {
    fun toModel(): QuestionReview {
        return QuestionReview(id, questionId, reviewerId, comment, rate, createdAt, isDeleted)
    }

    companion object {
        @JvmStatic
        fun from(questionReview: QuestionReview): QuestionReviewEntity {
            return QuestionReviewEntity(
                questionReview.id,
                questionReview.questionId,
                questionReview.reviewerId,
                questionReview.comment,
                questionReview.rate,
                questionReview.createdAt,
                questionReview.isDeleted
            )
        }
    }
}
